x-pack/test/functional/apps/dataset_quality/dataset_quality_privileges.ts

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { DatasetQualityFtrProviderContext } from './config';
+import { getInitialTestLogs, getLogsForDataset } from './data';
+
+export default function ({ getService, getPageObjects }: DatasetQualityFtrProviderContext) {
```

Comment:
If we write this whole test like this, it would make it much more easy to extend and readable. 

```
describe('Dataset quality handles user privileges', () => {
.....before () {
    // User creation
     1. create User will all priviledges
     2. create user will limited priviledges
     3. create user with no priviledges
    // Data generation
    1. Synthtrace data which can be used by all tests
    2. Install Integrations
}

....after() {
     1. Delete all created users
     2. Synthtrace clean
     3. Uninstall Integration
}

.....describe('All tests for 1st category of user lives here' ) {
   before () -> Login as 1st user
   after() -> Logout as 1st user
}
.....describe('All tests for 2nd category of user lives here' ) {
   before () -> Login as 2nd user
   after() -> Logout as 2nd user
}
.....describe('All tests for 3rd category of user lives here' ) {
   before () -> Login as 3rd user
   after() -> Logout as 3rd user
}
}

Reply:
I wanted to do like this initially but not all sub-suites need synthtrace data. Also creating the needed user per sub-suite makes sure other sub-suites are not affected if something goes wrong while creating/deleting a user, or if one sub-suites needs to be skipped.